data/data/aws/main.tf

```diff
@@ -1,7 +1,7 @@
 locals {
   private_endpoints = "${var.aws_endpoints == "public" ? false : true}"
   public_endpoints  = "${var.aws_endpoints == "private" ? false : true}"
-  private_zone_id   = "${var.aws_external_private_zone != "" ? var.aws_external_private_zone : join("", aws_route53_zone.int.*.zone_id)}"
+  private_zone_id   = "${join("", aws_route53_zone.int.*.zone_id)}"
```

Comment:
Do we know what this `join` is about?  I'd expect `"${aws_route53_zone.int.zone_id}"`.

Reply:
@wking The `join` is used to convert the list to a single value, knowing that the list will only have a single element. With the changes in this PR, the `join` is no longer necessary `aws_route53_zone` is no longer a collection of resources: it is a single resource.

Reply:
And to be clear, the code in the PR is what you are expecting `"${aws_route53_zone.int.zone_id}"`. Your comment is using an outdated version.

Reply:
Ah, I had been going through commit by commit.  Maybe you can squash the `join` removal further back.  Or just leave it, since it gets fixed by the end of the PR.

Reply:
I try to make the end result of each commit a working product. After that commit, the `join` is still needed since at that point `aws_route53_zone` is still a collection of resources.